point_pattern.py

```diff
@@ -119,7 +140,21 @@ def average_nearest_neighbor_distance(points):
      Measure of Spatial Relationships in Populations. Ecology. 35(4)
      p. 445-453.
     """
-    mean_d = 0
+    listOfDistances = []
+    
+    for n in points:
+        lowest = 100
+        for x in points:
+            if euclidean_distance(n,x)!= 0 and euclidean_distance(n,x) < lowest:
```

Comment:
What if the points are coincident (and valid)?  You might have a distance of 0.  You have two for loops, so you could check to see if the position (index) in the first for loop is the same as the index in the second.  Check out `enumerate` to help do this.